examples/java/src/main/java/org/apache/beam/examples/WindowedWordCount.java

```diff
@@ -106,7 +106,7 @@
     }
 
     @ProcessElement
-    public void processElement(ProcessContext c) {
+    public void processElement(@Element String element, OutputReceiver<String> receiver) {
```

Comment:
<!--thread_id:cc_183500920_t; commit:e7163d592d424a480ae9071c770022129d401905; resolved:0-->
<!--section:context-quote-->
> **jkff** wrote:
> OK so my main comment was that: the element parameter is nearly always present and always the most important parameter, so maybe we can elide the annotation for it, and say that it is by convention always the first parameter?
> 
> I saw one or two tests where the element parameter was missing, but I don't think that's common in production code, and it's easy to add an ignored parameter in those cases.

<!--section:body-->
I considered this initially, but I think this will be more error prone. Imagine cases where the element is an Instant and there is an Instant timestamp as well. We can always later allow eliding the @Element annotation if there is no such conflict, but IMO it's premature to do this now (I am not a fan of using parameter order to the function to determine this though)

Reply:
Alrighty then.